packages/docs/docs/search/chained-search.md

```diff
@@ -0,0 +1,110 @@
+  <Tabs groupId="language">
+    <TabItem value="ts" label="Typescript">
+      <MedplumCodeBlock language="ts" selectBlocks="chainedSearchTs">
+        {ExampleCode}
+      </MedplumCodeBlock>
+    </TabItem>
+    <TabItem value="cli" label="CLI">
+      <MedplumCodeBlock language="bash" selectBlocks="chainedSearchCli">
+        {ExampleCode}
+      </MedplumCodeBlock>
+    </TabItem>
+    <TabItem value="curl" label="cURL">
+      <MedplumCodeBlock language="bash" selectBlocks="chainedSearchCurl">
+        {ExampleCode}
+      </MedplumCodeBlock>
+    </TabItem>
+  </Tabs>
+</details>
+
+## Reverse Chained Search
+
+Reverse chained searches allow you to filter your results based on other resources that reference your results. This is done using the `_has` parameter.
+
+For example, you may want to search for any `Patient` resources that have had a certain code of an `Observation` made about them.
+
+<details><summary>Example: Search for any Patients that have had carbon dioxide observed in their blood</summary>
+  <Tabs groupId="language">
+    <TabItem value="ts" label="Typescript">
+      <MedplumCodeBlock language="ts" selectBlocks="reverseChainedSearchTs">
+        {ExampleCode}
+      </MedplumCodeBlock>
+    </TabItem>
+    <TabItem value="cli" label="CLI">
+      <MedplumCodeBlock language="bash" selectBlocks="reverseChainedSearchCli">
+        {ExampleCode}
+      </MedplumCodeBlock>
+    </TabItem>
+    <TabItem value="curl" label="cURL">
+      <MedplumCodeBlock language="bash" selectBlocks="reverseChainedSearchCurl">
+        {ExampleCode}
+      </MedplumCodeBlock>
+    </TabItem>
+  </Tabs>
+</details>
+
+In the above example `_has:Observation` filters for `Patient` resources that have an `Observation`. The `:patient` filters for `Observation` resources that reference a `patient` in the subject field. This is based on the `Observation` resource's `patient` search parameter. Finally, `:code=11557-6` filters for that specific code on the `Observation`.
+
+Reversed chained searches also allow you to to perform an "or" search by including multiple arguments as a comma separated list.
+
+<details><summary>Example: Search for any Patients that have had one of multiple observations</summary>
+  <Tabs groupId="language">
+    <TabItem value="ts" label="Typescript">
+      <MedplumCodeBlock language="ts" selectBlocks="reverseChainedOrSearchTs">
+        {ExampleCode}
+      </MedplumCodeBlock>
+    </TabItem>
+    <TabItem value="cli" label="CLI">
+      <MedplumCodeBlock language="bash" selectBlocks="reverseChainedOrSearchCli">
+        {ExampleCode}
+      </MedplumCodeBlock>
+    </TabItem>
+    <TabItem value="curl" label="cURL">
+      <MedplumCodeBlock language="bash" selectBlocks="reverseChainedOrSearchCurl">
+        {ExampleCode}
+      </MedplumCodeBlock>
+    </TabItem>
+  </Tabs>
+</details>
+
+### Nesting reverse chained searches
+
+It is also possible to nest the `_has` parameter.
```

Comment:
This is great, since `_has` is pretty confusing in this way.  This section could actually be generalized to cover the "mixing and matching" of chained parameters: forward and reverse chained params can be combined arbitrarily, e.g. `Patient?_has:Observation:subject:performer:CareTeam.participant:Practitioner.name=Bob` to select Patients with an Observation made by a care team containing a provider named Bob